### PR TITLE
Document how to stream JSON arrays

### PR DESCRIPTION
**Pull request by ehmicky**
This PR documents how to parse JSON array streams incrementally, as opposed to blocking the event loop once at the end using `JSON.parse()`.

This does not cover actual JSON streams formats (where each line is a single object, and there are no `[` `]` delimiters) although the utility mentioned supports those too.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -171,6 +171,21 @@ const {body: readableStream} = await fetch('https://example.com');
...
+[`getStreamAsArray()`](#getstreamasarraystream-options) can be combined with JSON streaming utilities to parse JSON incrementally.
+
+```js
+import {getStreamAsArray} from 'get-stream';
+import streamJson from 'stream-json';
+import streamJsonArray from 'stream-json/streamers/StreamArray.js';
+
+const stream = fs.createReadStream('big-array-of-objects.json', {encoding: 'utf8'});
+console.log(await getStreamAsArray(
+	stream.pipe(streamJson.parser()).pipe(streamJsonArray.streamArray()),
```
Comment (sindresorhus) [inline readme.md:185]: `pipeline` is generally recommended instead of pipe methods.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: readme.md (hunk 1 above)
Comment (ehmicky) [reply readme.md]: That's right! Good catch.

What about `compose()` instead? While `pipeline()` starts consuming the stream and returns a promise, `compose()` does not start consuming it and returns the stream.

Comment (sindresorhus) [reply readme.md]: Yeah, `compose` is probably the right choice. I've been too used to (apparently) incorrectly using `pipeline`.

File: readme.md [hunk 2]
```diff
@@ -215,6 +215,22 @@ const stream = fs.createReadStream('unicorn.txt');
 console.log(new Blob([await getStreamAsArrayBuffer(stream)]));
 ```
 
+### JSON streaming
+
+[`getStreamAsArray()`](#getstreamasarraystream-options) can be combined with JSON streaming utilities to parse JSON incrementally.
+
+```js
+import {compose} from 'node:stream';
```
Comment (sindresorhus) [inline readme.md:223]: `compose` feels too general as a name.
```suggestion
import {compose as composeStreams} from 'node:stream';
```

**Comment by ehmicky** [comment]
Fixed :+1:

**Pull request merged**